ase/sw/ase_ops.c

```diff
@@ -59,15 +59,14 @@ inline void set_loglevel(int level)
  * Generate unique socket server name
  * Generated name is populated in "name"
  */
-errno_t generate_sockname(char *name)
+int generate_sockname(char *name)
 {
-	errno_t err = EOK;
+	int res;
 
-	err = strncpy_s(name, strlen(SOCKNAME)+1, SOCKNAME,
-							strlen(SOCKNAME)+1);
-	if (err != EOK) {
-		ASE_ERR("%s: Error strncpy_s\n", __func__);
-		return err;
+	res = ase_strncpy(name, SOCKNAME, strlen(SOCKNAME)+1);
```

Comment:
If we are implementing our own version of these, could we follow the same parameter definitions as the safe_string call so that we perform identical checking?  I remember reading that baking your own string functions was frowned upon.  We'd probably raise fewer eyebrows if we could convincingly say that we ported the minimum set of safe_string fn's.

Reply:
I'll try that. It will result in less change, which may be an improvement.